scripts/docker/fedora.24/Dockerfile

```diff
@@ -1,6 +1,6 @@
 FROM fedora:24
 
-RUN dnf install -y bash git cmake wget which python clang-3.8.1-1.fc24.x86_64 llvm-devel-3.8.1-2.fc24.x86_64 make libicu-devel lldb-devel.x86_64 \
+RUN dnf install -y bash git cmake wget which python clang-3.8.0-1.fc24.x86_64 llvm-devel-3.8.0-1.fc24.x86_64 make libicu-devel lldb-devel.x86_64 \
```

Comment:
No, I'm not sure what the difference is between clang-3.8.0-2 (which was used until #1382 was merged earlier today) and clang-3.8.0-1.  It looks like release notes are just provided for the minor versions, e.g. for 3.8 at http://releases.llvm.org/3.8.0/tools/clang/docs/ReleaseNotes.html.

I don't think leaving it as 3.8.1-1 is a viable solution, because it will likely be removed at some point, just as 3.8.0-2 was.  The two options are using the stable version from the "fedora" package repo by specifying 3.8.0-1, and using the latest version from the "updates" package repo by not specifying a version at all. As mentioned in https://github.com/dotnet/core-setup/pull/1383#issuecomment-276447836, the main concern with not specifying a version is that we want to guarantee that the Docker image can be rebuilt in the future if needed and produce an image with the same components.

I ran a build locally with these changes (using the same steps as the official Fedora 24 build), and it succeeded.  Are there any functional regressions from the clang version that we have to be aware of that wouldn't be surfaced by a build failure?

/cc @dagood @MichaelSimons @dleeapho